### ISSUE TITLE
Authorization enforced with authentication is disabled

### ISSUE THREAD

**Issue by ayomawdb** [ballerina-platform/ballerina-lang#7381]
**Description:**

When scopes is defined for the service and authentication is disabled for a resource, resource access will try to enforce authorization even though authentication is disabled. 

**Steps to reproduce:**

Run following service and invoke "publicProfile" resource. 

```
import ballerina/http;

endpoint http:APIListener listener {
    port:9090
};

@http:ServiceConfig {
    authConfig:{
        authentication: { enabled: true },
        scopes:["students"]
    }
}
service<http:Service> student bind listener {

    @http:ResourceConfig {
        authConfig:{
            authentication: { enabled: false }
        }
    }
    publicProfile (endpoint client, http:Request req) {
        http:Response res = new;
        res.setStringPayload("Public Profile");
        _ = client -> respond(res);
    }
}

```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by isurulucky** [comment]
Fixed with PR https://github.com/ballerina-platform/ballerina-lang/pull/7423.

**Comment by anupama-pathirage** [comment]
Closing as it is fixed with #7423
